test-data/unit/check-newsemanal.test

```diff
@@ -268,3 +268,55 @@ x: T2
 reveal_type(x) # E: Revealed type is 'TypedDict('__main__.T2', {'x': builtins.str, 'y': builtins.int})'
 y: T4
 reveal_type(y) # E: Revealed type is 'TypedDict('__main__.T4', {'x': builtins.str, 'y': __main__.A})'
+
+[case testNewAnalyzerRedefinitionAndDeferral]
+# flags: --new-semantic-analyzer
+import a
+
+[file a.py]
+from b import x as y
+x = 0
+
+def y(): pass # E: Name 'y' already defined on line 2
+reveal_type(y) # E: Revealed type is 'builtins.int'
+
+y2 = y
+class y2: pass # E: Name 'y2' already defined on line 7
+reveal_type(y2) # E: Revealed type is 'builtins.int'
+
+y3, y4 = y, y
+from b import f as y3 # E: Incompatible import of "y3" (imported name has type "Callable[[], Any]", local name has type "int")
+reveal_type(y3) # E: Revealed type is 'builtins.int'
+
+[file b.py]
+from a import x
+
+def f(): pass
+
+[case testNewAnalyzerRedefinitionAndDeferral2]
+# flags: --new-semantic-analyzer
+import a
+
+[file a.py]
+from b import C as C2
+class C: pass
+
+class C2: pass # E: Name 'C2' already defined (possibly by an import)
+[file b.py]
+from a import C
+
+[case testNewAnalyzerImportStarForwardRef]
+# flags: --new-semantic-analyzer
+import a
+
+[file a.py]
+x: A
+reveal_type(x) # E: Revealed type is 'b.A'
+
+from b import *
+
+class A: pass # E: Name 'A' already defined (possibly by an import)
+
+[file b.py]
+class A: pass
+from a import x
```

Comment:
Maybe also add a test where a base class is start-imported in an import cycle? Something like:
```
[file a.py]
from b import *
class A(B): ...
[file b.py]
from a import A  # I am not sure about import here, may we need to try both orders of processing of modules
class B:
    x: A
```